solo-chains/node/tanssi-relay-service/src/dev_service.rs

```diff
@@ -719,7 +742,15 @@ fn new_full<
                                 &[],
                             )
                             .expect("Should be able to write to aux storage; qed");
+                        }
 
+                        if let Some(value) = upward_message.last() {
```

Comment:
I would take a different approach here, I would not sore it in an auxiliary message (as this is not a temporary thing we need to read). Instead, I would read all (and not only the last) message we send from this channel and inject it into the create function for `MockParachainsInherentDataProvider` as an argument

Reply:
We use a mock storage key when we want to store state, such as enable or disable some feature. Here we only want to send a single message for every time the rpc is called, right? Then we don't need storage, I believe passing the channel is ok.